src/SmartComponents/modules/reducers.js

```diff
@@ -54,7 +57,40 @@ function addSystemModalReducer(state = initialState, action) {
     }
 }
 
+function filterDropdownReducer(state = initialState, action) {
+    switch (action.type) {
+        case `${types.OPEN_FILTER_DROPDOWN}`:
+            return {
+                ...state,
+                filterDropdownOpened: !state.filterDropdownOpened
+            };
+
+        default:
+            return {
+                ...state
+            };
+    }
+}
+
+function filterByStateReducer(state = initialState, action) {
+    switch (action.type) {
+        case `${types.FILTER_BY_STATE}`:
+            return {
+                ...state,
+                stateFilter: action.payload.filter,
+                selectedFilter: action.payload.selected
```

Comment:
What do these two pieces of data represent? I needing which filter is selected, but what is the other piece of information being passed around?

Reply:
stateFilter is the filter that is used to compare against the comparison state text ("none", "SAME", "DIFFERENT", "No Data").

selectedFilter is used to display the selected option from the dropdown menu in the box after a selection has been made and the box collapsed.

Reply:
Wouldn't they be one in the same?

Reply:
I was handling them differently because one was "DIFFERENT" and the other was "Diff". I can restructure this so we don't have to over-complicate things.

Reply:
We could introduce a mapping so that there is the 'display' property and the 'value' property. Or go with Different for now.